packages/purpose-process/src/services/purposeService.ts

```diff
@@ -163,6 +165,16 @@ const retrievePurposeVersion = (
   return version;
 };
 
+const retrieveDraftPurposeVersion = (purpose: Purpose): PurposeVersion => {
+  const draftVersion = purpose.versions.find(
+    (v) => v.state === purposeVersionState.draft
+  );
+  if (draftVersion === undefined) {
+    throw purposeNotInDraftState(purpose.id);
```

Comment:
The error thrown here makes sense in the `updateDraftPurposeCreatedFromTemplate` function context, but if I use this function outside it, it may not. 
Here I'm looking for a Draft purpose version, and not checking if the purpose itself is in `Draft` state